### PR TITLE
[ci] Migrate unit tests and coverage to Github Actions

### LINKED ISSUE #1560: Move Travis CI testing to Github Actions

### PR DESCRIPTION
**Pull request by teojgo**
* Test also modules using dockerfiles.

Fixes #1560

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
+name: ReFrame CI
+
+on: [push, pull_request]
+
+jobs:
+  unittest:
+
```
Comment (vkarak) [inline .github/workflows/main.yml:7]: I would remove these lines.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install -r requirements.txt
```
Comment (vkarak) [inline .github/workflows/main.yml:22]: Why not with `./bootstrap.sh`?

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Setup Python for Coverage
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.8
+    - name: Coverage Calculation
```
Comment (vkarak) [inline .github/workflows/main.yml:37]:
```suggestion
    - name: Generate coverage report
```

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install -r requirements.txt
+    - name: Generic Unittests
+      run: |
+        ./test_reframe.py
+
+  coverage:
```
Comment (vkarak) [inline .github/workflows/main.yml:27]: Couldn't coverage just be part of the `unittests` stage above?

**Review by vkarak — CHANGES_REQUESTED** [review]

File: .github/workflows/main.yml (hunk 4 above)
Comment (teojgo) [reply .github/workflows/main.yml:27]: The coverage itself yes, but I don't know about the upload to codecov

**Review by vkarak — APPROVED** [review]
lgtm

**Pull request merged**